app/views/devise/shared/_links.html.erb

```diff
@@ -19,9 +19,10 @@
     <%= link_to t('.didn_t_receive_unlock_instructions'), new_unlock_path(resource_name) %><br />
   <% end -%>
 
-  <%- if devise_mapping.omniauthable? %>
+  <%- if devise_mapping.omniauthable? && AppConfig.facebook_login_enabled %>
```

Comment:
This is looking good. Next step is to add feature specs to prove that:
1. The FB link shows when enabled
2. The FB link doesn't show when not enabled
3. That the FB login works - both to add to existing user and to create a new user, if creating new users is enabled

This looks promising as a place to start: https://stackoverflow.com/questions/23050989/how-to-test-facebook-login-action-with-rails-omniauth-and-rspec

Reply:
Let me know how those look